openstack/orchestration/v1/stacks/requests.go

```diff
@@ -52,7 +52,6 @@ func (opts CreateOpts) ToStackCreateMap() (map[string]interface{}, error) {
 	if err := opts.TemplateOpts.getFileContents(opts.TemplateOpts.Parsed, ignoreIfTemplate, true); err != nil {
 		return nil, err
 	}
-	opts.TemplateOpts.fixFileRefs()
```

Comment:
Was that the only call to `fixFileRefs()` that was problematic, or should we drop more of them?
A quick grep shows more occurrences in the same file.

Reply:
Good point. I had looked into that at some point but because I didn't know what kind of manipulation was expected on the `Environment` part (just a few lines below, and where `getRRFileContents()` is used instead of `getFileContents()`), I decided to only touch the `Template` part that I was interested in and understood.
But I missed the fact that `Adopt`, `Update` and `Preview` also call `fixFileRefs()` for their template parts (after doing `getFileContents()`). I have now removed the calls from those other operations so they will also benefit from the fix.

Looking into `getRRFileContents()`, I saw it also uses `getFileContents()` but simply with a different filter function (`ignoreIfEnvironment()`). Hence, it is in fact already affected by my change since some of the files it loads could be re-generated with full paths expanded, as now done by `getFileContents()` without the need for `fixFileRefs()`.
So I went ahead and made a separate commit that completes similar changes for the env part and completely removes `fixFileRefs()`. I have however not tested this part other than with the included unit tests and am somewhat unsure as to whether it's okay to re-generate env content from the parsed structure as the code is now doing. Note that this was already the case with the previous commit (templates), just not for the top-level env content. I hope my explanation is not too unclear....

Reply:
Your explanation is perfectly clear, and I believe it is fine. It looks like a bug that was never noticed.